File: src/main/java/org/proshin/blog/socialnetwork/instagram/oauth/response/AccessToken.java
package org.proshin.blog.socialnetwork.instagram.oauth.response;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import lombok.Value;
import org.proshin.blog.socialnetwork.instagram.model.User;

@Value
public class AccessToken {

    String accessToken;
    User user;

    public AccessToken(String json) {
        JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
        this.accessToken = jsonObject.get("access_token").getAsString();
        this.user = new User(jsonObject.get("user").getAsJsonObject());
    }
}

File: src/main/kotlin/org/proshin/blog/socialnetwork/instagram/oauth/response/AccessToken.kt
package org.proshin.blog.socialnetwork.instagram.oauth.response

import com.github.salomonbrys.kotson.get
import com.github.salomonbrys.kotson.obj
import com.github.salomonbrys.kotson.string
import org.proshin.blog.socialnetwork.instagram.model.User

class AccessToken(json: String) {

    val accessToken: String
    val user: User

    init {
        val jsonElement = com.google.gson.JsonParser().parse(json)
        this.accessToken = jsonElement["access_token"].string
        this.user = User(jsonElement["user"].obj)
    }
}